public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -363,7 +368,7 @@ export class CloudWatchDatasource
         }
       }),
       map(([dataFrames, failedAttempts]) => {
-        if (failedAttempts >= MAX_ATTEMPTS) {
+        if (timeoutFunc()) {
```

Comment:
I don't think we need to evaluate again the timeout since it was done already in `runWithRetry`. What we can do here is evaluate the error that that function returns. If the error message `.contains("Some queries timed out")` then you handle the timeout accordingly here.

Reply:
This is a separate api call (`GetQueryResults` on line 330) from the runWithRetry call to `StartQuery`. Checking the error for `StartQuery` won't tell us if we can do more `GetQueryResults` calls or not.

Reply:
ah, I think I get it now. If I understood it properly: 

Before this PR, `StartQuery` started a timeout and if everything went fine (the timeout is not reached), it returned empty Frames (with some metadata). Then `GetQueryResults`, with the info returned by `StartQuery` started a second timeout and if that timeout was reached or if after 5 retries it didn't get any update, then it failed.

In this PR, you have removed the second timeout and the max-retries condition.

I am not sure if `StartQuery` really needs a timeout but if before it had it, we are probably safer leaving it.

A minor thing, to simplify things, would be to delete the `timeoutFunction` definition in the `runWithRetry` function. The only place in which that function is used is here and you are already defining the timeout function outside `runWithRetry`. As far as I can see, you can remove `options` from that function and make the `timeoutFunction` mandatory. WDYT?

Reply:
done